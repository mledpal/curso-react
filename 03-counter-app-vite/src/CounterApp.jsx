import { useState } from  'react';
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( value );

    const sumarContador = () => setCounter( counter + 1 );
    const restarContador = () => setCounter( counter - 1 );
    const resetContador = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter } </h2>
            <button onClick={ sumarContador }> +1 </button>
            <button onClick={ restarContador }> -1 </button>
            <button onClick={ resetContador }> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}