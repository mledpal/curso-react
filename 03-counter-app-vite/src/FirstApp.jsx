import PropTypes from 'prop-types';

const getSaludo = (nombre) =>  'Hola '+ nombre

export const FirstApp = ( { titulo, subtitulo, nombre } ) => {

    // if(!titulo) {
    //     throw new Error('El título no existe');
    // }

    return (
        <>
        <h1>{ getSaludo('Miguel') }</h1>
        {/* <code>{ JSON.stringify (newMessage) }</code> */}
        <p>{ titulo }</p>
        <p>{ subtitulo }</p>
        <p>{ nombre }</p>
        </>
    );
}

FirstApp.propTypes = {
    nombre: PropTypes.string,
    subtitulo: PropTypes.number,
    titulo: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    nombre: 'Miguel Ledesma',
    subtitulo: 0,
    titulo : 'No hay título',
}