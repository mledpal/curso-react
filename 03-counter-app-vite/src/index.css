* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;    
}

body {
    align-items: center;
    background-image: linear-gradient(90deg, #0d4da0, #4f96cf);
    color: #eee;
    display: flex;
    flex-direction: column;
    font-family: System-ui;
    font-weight: 600;
    justify-content: center;
    min-height: 100vh;
    padding: 20px;
    width: 100%;
    text-align: center;
}

h1 {
    font-size: 2em;
}

button {
    padding: 10px 30px;
    border-radius: 5px;
    box-shadow: 1px 1px 4px #222, -1px -1px 4px #555 inset;
    font-size: 1.3rem;
    margin: 5px;
}