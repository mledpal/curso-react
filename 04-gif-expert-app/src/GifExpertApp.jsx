import { useState } from "react";
import { AddCategory} from "./components/AddCategory";
import { GiftGrid } from "./components/GifGrid";


export const GifExpertApp = () => { 

    const [categories, setCategories] = useState(['Dragon Ball']);

    const onAddCategory = ( nuevaCategoria ) => {
        //setCategories( ( cat ) => [...cat, 'Call Of Duty'] );
  
        if(!categories.includes(nuevaCategoria)){
            setCategories( [nuevaCategoria, ...categories] );
        } else {
            console.log('Ya existe');
        }
        
    };

    return (
        <>
            
            <h1>GifExpertApp</h1>

            <AddCategory onAddCategory = { onAddCategory }/>
            
            { categories.map( category => ( 
                <GiftGrid 
                    key={ category } 
                    category={ category } 
                /> 
            ))}
        </>
    )

}