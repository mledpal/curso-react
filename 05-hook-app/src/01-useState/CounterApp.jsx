import { useState } from "react";

export const CounterApp = () => {
	const [state, setCounter] = useState({
		counter1: 10,
		counter2: 20,
		counter3: 30,
	});

	const { counter1, counter2, counter3 } = state;

	const increment = () => {
		setCounter({
			...state,
			counter1: counter1 + 1,
		});
	};

	const decrement = () => {
		setCounter({
			...state,
			counter1: counter1 - 1,
		});
	};

	const reset = () => {
		setCounter({
			counter1: 10,
			counter2: 20,
			counter3: 30,
		});
	};

	return (
		<>
			<h1>Counter1 : {counter1} </h1>
			<h1>Counter2 : {counter2} </h1>
			<h1>Counter3 : {counter3} </h1>
			<hr />

			<button onClick={increment} className='btn btn-primary'>
				+1
			</button>

			<button onClick={reset} className='btn btn-secondary'>
				Reset
			</button>

			<button onClick={decrement} className='btn btn-secondary'>
				-1
			</button>
		</>
	);
};
