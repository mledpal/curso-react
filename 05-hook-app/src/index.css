* { 
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  width: 100vw;
  max-width: 100%;
  min-height: 100vh;
  font-family: system-ui;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

button {  
  padding: 20px;
  margin: 10px;
}