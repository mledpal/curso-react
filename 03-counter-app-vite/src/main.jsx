/* eslint-disable react-refresh/only-export-components */
/* eslint-disable no-unused-vars */

import React from 'react';
import ReactDOM from 'react-dom/client';

// import { App } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';

import { CounterApp } from './CounterApp';

import './index.css';

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>        
        <CounterApp value = { 0 }/> 
    </React.StrictMode>
)