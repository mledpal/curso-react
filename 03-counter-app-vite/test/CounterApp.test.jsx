/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */

import { CounterApp } from '../src/CounterApp';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Pruebas con el componente CounterAPP', () => { 

    const valor = 100;

    test('Debe hacer match con el snapshot', () => { 
        const { container } = render(<CounterApp value={ valor } />);
        expect(container).toMatchSnapshot();   
    });

    test('Debe mostrar el valor por defecto de 100', () => { 

        render(<CounterApp value ={ valor } />);
        expect(screen.getByText(valor)).toBeTruthy();       
        // expect(screen.getByRole('heading', { level : 2 }).innerHTML).toContain(`${valor}`);
    });


    test('Debe incrementar con el boton +1', () => { 
        render(<CounterApp value ={ valor } />);
        
        fireEvent.click(screen.getByText('+1'));

        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(`${valor + 1}`);
        
    });

    test('Debe decrementar con el boton -1', () => { 
        render(<CounterApp value ={ valor } />);
        
        fireEvent.click(screen.getByText('-1'));

        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(`${valor -1 }`);

    });

    test('Debe resetear el valor con el boton reset', () => {
        render(<CounterApp value ={ valor } />);
        
        fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByText('-1'));

        fireEvent.click(screen.getByRole('button', { name: 'btnReset' }));
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(`${valor}`);
    });

});