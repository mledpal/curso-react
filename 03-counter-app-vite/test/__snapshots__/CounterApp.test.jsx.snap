// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente CounterAPP Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btnReset"
  >
     Reset 
  </button>
</div>
`;
